sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -2282,11 +2285,23 @@ public <T> Mono<FeedResponse<T>> readMany(
                                     // TODO: this does double serialization: FIXME
                                     finalList.addAll(page.getResults().stream().map(document ->
                                         ModelBridgeInternal.toObjectFromJsonSerializable(document, klass)).collect(Collectors.toList()));
+                                    aggregateRequestStatistics.addAll(BridgeInternal.getClientSideRequestStatisticsList(page.getCosmosDiagnostics()));
+                                    if (firstQueryPlanDiagnosticContext == null)
```

Comment:
I don't think we have `queryPlan` calls for readMany. So this might not be required?
@xinlian12 / @jeet1995  can you please confirm this?

Reply:
Yeah, only further review, it looks like passing null for this is allowed, so I'll remove this part.

Reply:
Done in 935c70d2c97

Reply:
@kushagraThapar No queryPlanDiagnostics are set on the feedResponses we get from the upstream.